src/main/java/com/cloudbees/jenkins/GitHubTriggerEvent.java

```diff
@@ -0,0 +1,125 @@
+package com.cloudbees.jenkins;
+
+import javax.servlet.http.HttpServletRequest;
+import jenkins.scm.api.SCMEvent;
+
+/**
+ * Encapsulates an event for {@link GitHubTrigger2}.
+ *
+ * @since TODO
+ */
+public class GitHubTriggerEvent {
+
+    /**
+     * The timestamp of the event (or if unavailable when the event was received)
+     */
+    private final long timestamp;
+    /**
+     * The origin of the event (see {@link SCMEvent#originOf(HttpServletRequest)})
+     */
+    private final String origin;
+    /**
+     * The user that the event was provided by.
+     */
+    private final String triggeredByUser;
+
+    private GitHubTriggerEvent(long timestamp, String origin, String triggeredByUser) {
+        this.timestamp = timestamp;
+        this.origin = origin;
+        this.triggeredByUser = triggeredByUser;
+    }
+
+    public static Builder create() {
+        return new Builder();
+    }
+
+    public long getTimestamp() {
+        return timestamp;
+    }
+
+    public String getOrigin() {
+        return origin;
+    }
+
+    public String getTriggeredByUser() {
+        return triggeredByUser;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        GitHubTriggerEvent that = (GitHubTriggerEvent) o;
+
+        if (timestamp != that.timestamp) {
+            return false;
+        }
+        if (origin != null ? !origin.equals(that.origin) : that.origin != null) {
+            return false;
+        }
+        return triggeredByUser != null ? triggeredByUser.equals(that.triggeredByUser) : that.triggeredByUser == null;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = (int) (timestamp ^ (timestamp >>> 32));
+        result = 31 * result + (origin != null ? origin.hashCode() : 0);
+        result = 31 * result + (triggeredByUser != null ? triggeredByUser.hashCode() : 0);
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return "GitHubTriggerEvent{" +
+                "timestamp=" + timestamp +
+                ", origin='" + origin + '\'' +
+                ", triggeredByUser='" + triggeredByUser + '\'' +
+                '}';
+    }
+
+    /**
+     * Builder for {@link GitHubTriggerEvent} instances..
+     */
+    public static class Builder {
```

Comment:
Because then you cannot have an equals method that compares fields and you cannot trust consumers of the event not to mutate. Immutable is best (I don't like `@DataBoundSetter` BTW)